src/config/eww_config.rs

```diff
@@ -36,35 +37,29 @@ impl EwwConfig {
     }
 
     pub fn read_from_file<P: AsRef<std::path::Path>>(path: P) -> Result<Self> {
-        let content = util::replace_env_var_references(std::fs::read_to_string(path)?);
+        let content = util::replace_env_var_references(std::fs::read_to_string(path.as_ref())?);
         let document = roxmltree::Document::parse(&content)?;
 
-        let result = EwwConfig::from_xml_element(XmlNode::from(document.root_element()).as_element()?.clone());
+        let result = EwwConfig::from_xml_element(XmlNode::from(document.root_element()).as_element()?.clone(), path.as_ref());
         result
     }
 
-    pub fn from_xml_element(xml: XmlElement) -> Result<Self> {
-
-        // TODO: This is not the way
-        let CONFIG_DIR: std::path::PathBuf = std::env::var("XDG_CONFIG_HOME")
-            .map(|v| PathBuf::from(v))
-            .unwrap_or_else(|_| PathBuf::from(std::env::var("HOME").unwrap()).join(".config"))
-            .join("eww");
+    pub fn from_xml_element<P: AsRef<std::path::Path>>(xml: XmlElement, path: P) -> Result<Self> {
 
         // !!! This doesnt seem that bad
         let includes =
             match xml.child("includes") {
                 Ok(tag) => tag.child_elements()
                     .map(|child| {
```

Comment:
Please also verify the tag name here, the same way it is done for other configuration tags. This is to avoid that the user is able to put _any_ tag, no matter it's name in here, as long as it has the correct "path" attribute

Reply:
Done tho